Step 3: Add support for return_counts to torch.unique for dim not None

zasdfgbnm opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #18661 Step 7: remove _unique
* #18655 Step 6: Rename _unique2 to unique and add int? dim
* #18654 Step 5: remove _unique_dim in favor of unique_dim
* #18651 Step 4: add support for unique with dim=None
* **#18650 Step 3: Add support for return_counts to torch.unique for dim not None**
* #18649 Step 2: Rename _unique_dim2_temporary_will_remove_soon to unique_dim

See step 1 for the description of the full stack

Differential Revision: [D14892319](https://our.internmc.facebook.com/intern/diff/D14892319)

zasdfgbnm committed:
Step 3: Add support for return_counts to torch.unique for dim not None

See step 1 for the description of the full stack

VitalyFedyunin commented on torch/functional.py:
```diff
@@ -433,20 +441,28 @@ def unique(input, sorted=True, return_inverse=False, dim=None):
     """
     if dim is not None:
-        output, inverse_indices = torch._unique_dim(
+        output, inverse_indices, counts = torch._C._VariableFunctions.unique_dim(
```
maybe torch.unique_dim ?

zasdfgbnm replied:
@VitalyFedyunin I don't think `torch.unique_dim`, because we don't want to expose `unique_dim` to Python, because we already have `unique(dim=...)`. See:
https://github.com/pytorch/pytorch/pull/18649/files#diff-61b893a141a037c275f12eba9a9b864fR388

zasdfgbnm commented on torch/tensor.py:
```diff
@@ -343,23 +343,7 @@ def unique(self, sorted=True, return_inverse=False, dim=None):
-            return output, inverse_indices
-        else:
-            return output
+        return torch.unique(self, sorted=sorted, return_inverse=return_inverse, dim=dim)
```
I seem to forget adding `return_counts` here. Will fix soon.

zasdfgbnm replied:
fixed

zasdfgbnm committed:
Update on "Step 3: Add support for return_counts to torch.unique for dim not None"

Step 3: Add support for return_counts to torch.unique for dim not None

See step 1 for the description of the full stack

gh-metadata: pytorch pytorch 18650 gh/zasdfgbnm/3/head

zasdfgbnm committed:
Update on "Step 3: Add support for return_counts to torch.unique for dim not None"

Step 3: Add support for return_counts to torch.unique for dim not None

See step 1 for the description of the full stack

gh-metadata: pytorch pytorch 18650 gh/zasdfgbnm/3/head

zasdfgbnm committed:
Update on "Step 3: Add support for return_counts to torch.unique for dim not None"

Step 3: Add support for return_counts to torch.unique for dim not None

See step 1 for the description of the full stack

gh-metadata: pytorch pytorch 18650 gh/zasdfgbnm/3/head

zasdfgbnm committed:
Update on "Step 3: Add support for return_counts to torch.unique for dim not None"

Step 3: Add support for return_counts to torch.unique for dim not None

See step 1 for the description of the full stack

gh-metadata: pytorch pytorch 18650 gh/zasdfgbnm/3/head

zasdfgbnm committed:
Update on "Step 3: Add support for return_counts to torch.unique for dim not None"

Step 3: Add support for return_counts to torch.unique for dim not None

See step 1 for the description of the full stack

gh-metadata: pytorch pytorch 18650 gh/zasdfgbnm/3/head

zasdfgbnm committed:
Update on "Step 3: Add support for return_counts to torch.unique for dim not None"

Step 3: Add support for return_counts to torch.unique for dim not None

See step 1 for the description of the full stack

gh-metadata: pytorch pytorch 18650 gh/zasdfgbnm/3/head

zasdfgbnm commented:
@VitalyFedyunin This should also be ready (if tests don't fail).

zasdfgbnm committed:
Update on "Step 3: Add support for return_counts to torch.unique for dim not None"

Step 3: Add support for return_counts to torch.unique for dim not None

See step 1 for the description of the full stack

gh-metadata: pytorch pytorch 18650 gh/zasdfgbnm/3/head

zasdfgbnm commented:
@VitalyFedyunin After merging step 2, do I need to rebase this, or it is fine?

VitalyFedyunin commented:
It is fine, there are two test failing internally. I plan to look at them closely tonight (might be unrelated thing).
